youtube_dl/extractor/rai.py

```diff
@@ -500,3 +502,108 @@ def _real_extract(self, url):
         info.update(relinker_info)
 
         return info
+
+
+class HTMLListAttrsParser(compat_HTMLParser):
+    def __init__(self):
+        compat_HTMLParser.__init__(self)
+        self.items = []
+        self._level = 0
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'li' and self._level == 0:
+            self.items.append(dict(attrs))
+        self._level += 1
+
+    def handle_endtag(self, tag):
+        self._level -= 1
+
+
+class RaiPlayRadioBaseIE(InfoExtractor):
+    _BASE = 'https://www.raiplayradio.it'
+
+    def parse_list(self, webpage):
+        parser = HTMLListAttrsParser()
+        parser.feed(webpage)
+        parser.close()
+        return parser.items
+
+    def get_playlist_iter(self, url):
+        webpage = self._download_webpage(url, url)
```

Comment:
`url` is not a video id.

Reply:
Using the id extracted from the URL. Hope this is enough.